### PR TITLE
Create methods and impement CRUD methods

### PR DESCRIPTION
**Pull request by Daf98**
#### Create models
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
#### Use models to insert data
- Create at least one user by running the following code:
   `first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')`
 `second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')`
- Create at least 4 posts written by one of the users you created by running the following code:
`first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')`
- Create at least 6 posts comments for one of the posts you created by running the following code:
`Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )`
##### Users
- A method that returns the 3 most recent posts for a given user.
##### Posts
- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.
##### Comments
- A method that updates the comments counter for a post.
##### Likes
- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (VICTOR-CHUKWU) [inline app/models/comment.rb]: - [x] According to the project requirement, kindly change user to author hence the foreign keys would be author_id, you can change this from the migration file so this line should look like this `belongs_to :author, class_name: 'User'`

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (VICTOR-CHUKWU) [inline app/models/like.rb]: - [x] as mentioned above kindly change to author

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  belongs_to :user
```
Comment (VICTOR-CHUKWU) [inline app/models/post.rb]: - [x] as mentioned above, kindly change to author

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Daf98  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi @Daf98, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**